XenAdmin/Commands/SuspendVMCommand.cs

```diff
@@ -201,10 +201,18 @@ protected override string GetCantExecuteReasonCore(SelectedItem item)
             {
                 return Messages.VM_SHUT_DOWN;
             }
-            else if (vm.power_state == vm_power_state.Suspended)
+            if (vm.power_state == vm_power_state.Suspended)
             {
                 return Messages.VM_ALREADY_SUSPENDED;
             }
+            if (vm.HasGPUPassthrough())
+            {
+                return FriendlyErrorNames.VM_HAS_PCI_ATTACHED;
+            }
+            if (vm.allowed_operations != null && !vm.allowed_operations.Contains(vm_operations.suspend))

```

Comment:
With this change, GetCantExecuteNoToolsOrDriversReasonCore(item) will never be reached, so we will never see the "tools not installed" reason.

Reply:
I will move the GetCantExecuteNoToolsOrDriversReasonCore(item) call above the newer checks.